src/main/java/io/cdap/plugin/snowflake/sink/batch/SnowflakeBatchSink.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.snowflake.sink.batch;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.batch.Output;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.dataset.lib.KeyValue;
+import io.cdap.cdap.etl.api.Emitter;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.StageConfigurer;
+import io.cdap.cdap.etl.api.batch.BatchRuntimeContext;
+import io.cdap.cdap.etl.api.batch.BatchSink;
+import io.cdap.cdap.etl.api.batch.BatchSinkContext;
+import io.cdap.plugin.common.LineageRecorder;
+import org.apache.hadoop.io.NullWritable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.stream.Collectors;
+
+/**
+ * Plugin inserts records into Snowflake tables.
+ */
+@Plugin(type = BatchSink.PLUGIN_TYPE)
+@Name(SnowflakeBatchSink.PLUGIN_NAME)
+@Description("Writes records to Snowflake")
+public class SnowflakeBatchSink extends BatchSink<StructuredRecord, NullWritable, CSVRecord> {
+  private static final Logger LOG = LoggerFactory.getLogger(SnowflakeBatchSink.class);
+
+  public static final String PLUGIN_NAME = "Snowflake";
+
+  private final SnowflakeSinkConfig config;
+  private StructuredRecordToCSVRecordTransformer transformer;
+
+  public SnowflakeBatchSink(SnowflakeSinkConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    super.configurePipeline(pipelineConfigurer);
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    config.validate(stageConfigurer.getInputSchema(), stageConfigurer.getFailureCollector());
```

Comment:
Already have this code here:
```
  private void validateInputSchema(Schema schema, FailureCollector failureCollector) {
    // schema can be null in case it is a macro
    if (schema == null) {
      return;
    }
   ...
```
I tested it too. Works fine.